src/utils/request.js

```diff
@@ -0,0 +1,70 @@
+const fetch = require('node-fetch');
+
+/**
+ * @typedef {import('node-fetch').Response} NodeFetchResponse
+ */
+
+/* Utility class to manage API requests */
+class Request {
+  /**
+   * Create an instance of Request
+   * @param {string} baseUrl Warehouse API base url
+   * @param {string} opts.username Username
+   * @param {string} opts.password Password
+   */
+  constructor({ baseUrl, username, password }) {
+    this._baseUrl = baseUrl;
+    this._auth = this._basicAuth({ username, password });
+  }
+
+  /**
+   * Validate http response
+   * @param {NodeFetchResponse} res HTTP response
+   * @returns {Promise<void>} Promise rappresenting the verification result
+   */
+  async _checkRespStatus(res) {
+    if (!res.ok) {
+      const errorBody = await res.text();
+      throw new Error(
+        `HTTP Error Response: ${res.status} ${res.statusText} ${errorBody}`
+      );
+    }
+  }
+
+  /**
+   * Generate basic authorization header
+   * @private
+   * @param {Object} opts Method paramaters
+   * @param {string} opts.username Username
+   * @param {string} opts.password Password
+   * @returns {string} Basic auth header
+   */
+  _basicAuth({ username, password }) {
+    const b64 = Buffer.from(`${username}:${password}`).toString('base64');
+    return `Basic ${b64}`;
+  }
+
+  /**
+   * Send a POST http request
+   * @param {string} endpoint API endpoint
+   * @param {Object} body JSON-like body data
+   * @returns {Promise<Object>} Promise rappresenting the API response
+   */
+  async post(endpoint, body) {
+    /** @type NodeFetchResponse */
+    const resp = await fetch(`${this._baseUrl}${endpoint}`, {
```

Comment:
we probably want to use a new promise based API client at the core here so we don't have to implement that twice? What do you think? I guess you could set that up in here and eventually pull it out into the api-client module when we need to integrate it elsewhere.

Reply:
Are you suggesting to have a js sdk to interact with the API?

Reply:
If so, I am still thinking about how we should do this.

Reply:
I feel the way I am going to deal with this is: implementing http calls here for now, then refactor later and eventually extract the code here into the client module.

Reply:
FYI, I am still considering to use swagger to generate the client entirely. Haven't got enough time to play around with that yet.